### PR TITLE
[receivers/mongodbatlasrecevier] fix first scrape lookback time

### PR DESCRIPTION
**Pull request by mstumpfx**
**Description:** The first `start` lookback was being set to the 0 value of time.Time, and ultimately being passed to the mongodb atlas api with a value of `0001-01-01T00:00:00Z`, which is `bad`. Results in an attempt to query a large amount of data.

### CODE REVIEW

**Review by anuraaga — COMMENTED** [review]
Can you add a unit test to verify the behavior?

**Comment by mstumpfx** [comment]
> Can you add a unit test to verify the behavior?
@anuraaga I added a attempted to split out the functionality to make it easier to test, and added `receiver_test.go` with a unit test for it.

File: receiver/mongodbatlasreceiver/receiver_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	}{
+		{
+			name: "initial lookback is now() - collection_interval",
+			run: func(t *testing.T) {
+				now := time.Now()
+				tc := recv.timeConstraints(now)
+				require.NotNil(t, tc)
+				require.Equal(t, tc.start, now.Add(cfg.CollectionInterval*-1).UTC().Format(time.RFC3339))
+				// set lookback for next test after this one is done
+				recv.lastRun = now
```
Comment (dmitryax) [inline receiver/mongodbatlasreceiver/receiver_test.go:69]: It doesn't seem right to make one test case dependent on another. I would recommend extracting subtests into a plain test case or making them independent.

**Review by dmitryax — COMMENTED** [review]
Just one nit. otherwise LGTM

File: receiver/mongodbatlasreceiver/receiver_test.go (hunk 1 above)
Comment (mstumpfx) [reply receiver/mongodbatlasreceiver/receiver_test.go:69]: I was thinking the same thing when I wrote it tbh, I'll split it out now.

Comment (mstumpfx) [reply receiver/mongodbatlasreceiver/receiver_test.go]: @dmitryax updated to make them independent.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**